src/main/java/seedu/address/logic/parser/RemarkCommandParser.java

```diff
@@ -0,0 +1,33 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.RemarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.patient.Remark;
+
+public class RemarkCommandParser implements Parser<RemarkCommand> {
+
+    @Override
+    public RemarkCommand parse(String userInput) throws ParseException {
+        requireNonNull(userInput);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(userInput,
+                PREFIX_REMARK);
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
```

Comment:
I think this should catch `ParseException` since the `parseIndex` throws it and no mention about `IllegalValueException`

Reply:
Oh I've done some tracing and found that `IllegalValueException` is normally used when your method is passed a wrong input.  

In the case of `parseIndex`, a `ParseException` is thrown to represent the errors encountered when the input index isn't a valid number (i.e >0) or is a letter. 

However, since `ParseException` extends `IllegalValueException`, using `IllegalValueException` in this catch block will allow us to catch various kinds of `ParseException` if we decide to extend the functionality of our code in the future since it's the parent class.

Reply:
Hmm makes sense, alright I'll change it over to `ParseException`! Figured otherwise at first as the tutorial employed the use of `IllegalValueException` as well.

Reply:
Ohh I was comparing it with the other command parsers and they were using `ParseException`. So shd be ok to change it to `ParseException`

Reply:
Changed it! 👍🏻